frame/support/procedural/src/benchmark.rs

```diff
@@ -0,0 +1,816 @@
+			let mut extrinsic_call: Option<ExtrinsicCallDef> = None;
+			if let Stmt::Semi(Expr::Call(expr_call), _semi) = child {
+				// non-block (encoded) case
+				for (k, attr) in (&expr_call.attrs).iter().enumerate() {
+					if let Some(segment) = attr.path.segments.last() {
+						if let Ok(_) = syn::parse::<keywords::extrinsic_call>(
+							segment.ident.to_token_stream().into(),
+						) {
+							let mut expr_call = expr_call.clone();
+
+							// consume #[extrinsic_call] tokens
+							expr_call.attrs.remove(k);
+
+							// extract origin from expr_call
+							let origin = match expr_call.args.first() {
+								Some(arg) => arg.clone(),
+								None =>
+									return Err(Error::new(
+										expr_call.args.span(),
+										"Single-item extrinsic calls must specify their origin as the first argument.",
+									)),
+							};
+							extrinsic_call = Some(ExtrinsicCallDef::Encoded { origin, expr_call });
+						}
+					}
+				}
+			} else if let Stmt::Expr(Expr::Block(block)) = child {
+				// block case
+				for (k, attr) in (&block.attrs).iter().enumerate() {
+					if let Some(segment) = attr.path.segments.last() {
+						if let Ok(_) = syn::parse::<keywords::extrinsic_call>(
+							segment.ident.to_token_stream().into(),
+						) {
+							let mut block = block.clone();
+
+							// consume #[extrinsic_call] tokens
+							block.attrs.remove(k);
+
+							extrinsic_call = Some(ExtrinsicCallDef::Block(block));
+						}
+					}
+				}
+			}
+
+			// return parsed BenchmarkDef
+			if let Some(extrinsic_call) = extrinsic_call {
+				return Ok(BenchmarkDef {
+					params,
+					setup_stmts: Vec::from(&item_fn.block.stmts[0..i]),
+					extrinsic_call,
+					verify_stmts: Vec::from(
+						&item_fn.block.stmts[(i + 1)..item_fn.block.stmts.len()],
+					),
+					extra,
+					skip_meta,
+				})
+			}
+			i += 1;
+		}
+		return Err(Error::new(
+			item_fn.block.brace_token.span,
+			"No valid #[extrinsic_call] annotation could be found in benchmark function body.",
+		))
+	}
+}
+
+/// Parses and expands a `#[benchmarks]` or `#[instance_benchmarks]` invocation
+pub fn benchmarks(attrs: TokenStream, tokens: TokenStream, instance: bool) -> TokenStream {
+	let module = parse_macro_input!(tokens as ItemMod);
+	let where_clause = match syn::parse::<Nothing>(attrs.clone()) {
+		Ok(_) => quote!(),
+		Err(_) => parse_macro_input!(attrs as WhereClause).predicates.to_token_stream(),
+	};
+	let mod_vis = module.vis;
+	let mod_name = module.ident;
+	let mut expanded_stmts: Vec<TokenStream2> = Vec::new();
+	let mut benchmark_defs: Vec<BenchmarkDef> = Vec::new();
+	let mut benchmark_names: Vec<Ident> = Vec::new();
+	let mut extra_benchmark_names: Vec<Ident> = Vec::new();
+	let mut skip_meta_benchmark_names: Vec<Ident> = Vec::new();
+	let mut args: Option<BenchmarkAttrs> = None;
+	if let Some(mut content) = module.content {
+		for stmt in &mut content.1 {
+			let mut found_item: Option<ItemFn> = None;
+			if let Item::Fn(func) = stmt {
```

Comment:
Let's use the new let-else syntax here to reduce rightward drift:

```suggestion
			let Item::Fn(func) = stmt {
			//...
			} else { continue; }
```

Reply:
Here and other places in the code too where you used if-lets.